### PR TITLE
Add Codespaces container

### PR DESCRIPTION
**Pull request by carlosmartinez**
## Context

We (and particularly designers and content designers) would like to be able to see their changes in realtime. Github Codespaces is a good option for this.

## Changes proposed in this pull request

Set up a Codespaces devcontainer.

## Link to Trello card

https://trello.com/c/BkMjsgo3/1138-create-docker-image-for-codespaces

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by JR-G — APPROVED** [review]
🦑

File: .devcontainer/boot.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+echo "Setting SSH password for vscode user..."
```
Comment (frankieroberto) [inline .devcontainer/boot.sh:3]: Nitpick:

```suggestion
echo "Setting SSH password for VS Code users..."
```

File: docs/codespaces.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+## Modifying data in a Rails console
+
+- Open a new terminal (`+` button)
+- Run `bundle exec rails console`
+- Now you can do things like `ApplicationForm.first.update(first_name: 'Falafel')`
+
+## Gotchas
+
+- The codespace will send emails with application URLs relative to http://localhost:3000. Thus for example when you receive a magic link, you'll need to copy the URL and change the host to whatever Codespaces has generated for you.
```
Comment (frankieroberto) [inline docs/codespaces.md:32]:
```suggestion
- The codespace will send emails with application URLs relative to http://localhost:3000. For example, when you receive a magic link, you'll need to copy the URL and change the host to whatever Codespaces has generated for you.
```

**Review by frankieroberto — APPROVED** [review]
🙌  Amazing stuff! 👏

File: docs/codespaces.md (hunk 2 above)
Comment (carlosmartinez) [reply docs/codespaces.md:32]: correct my slightly pompous use of the word 'thus' = insult my whole family

**Pull request merged**